WordPress/WordPressApi/WordPressOrgXMLRPCApi.swift

```diff
@@ -0,0 +1,180 @@
+                    success: SuccessResponseBlock,
+                    failure: FailureReponseBlock) -> NSProgress?
+    {
+        let progress = NSProgress()
+        progress.totalUnitCount = 1
+        let xmlRPCParameters = XMLRPCCallParameters(method: method, parameters: parameters)
+        let task = sessionManager.POST(apiBaseURLString, parameters: xmlRPCParameters, success: { (dataTask, result) in
+            success(responseObject: result, httpResponse: dataTask.response as? NSHTTPURLResponse)
+            progress.completedUnitCount = 1
+            }, failure: { (dataTask, error) in
+                failure(error: error, httpResponse: dataTask?.response as? NSHTTPURLResponse)
+                progress.completedUnitCount = 1
+            }
+        )
+        if let task = task {
+            progress.cancellationHandler = {
+                task.cancel()
+            }
+            return progress
+        } else {
+            return nil
+        }
+    }
+}
+
+private final class XMLRPCCallParameters {
+    let method: String
+    let parameters: [AnyObject]?
+    let streamingCacheFile: NSURL?
+
+    init(method:String, parameters:[AnyObject]? = nil, streamingCacheFile: NSURL? = nil) {
+        self.method = method
+        self.parameters = parameters
+        self.streamingCacheFile = streamingCacheFile
+    }
+}
+
+/**
+ Error constants for the WordPress XMLRPC API
+
+ - CallNeedtoBePOST:                   The parameters sent to the server where invalid
+ - RequestSerializationFailed:     The serialization of the request failed
+ - Unknown:                        Unknow error happen
+ */
+@objc public enum WordPressOrgXMLRPCApiError: Int, ErrorType {
+    case CallNeedsToBePOST
+    case InvalidArguments
+    case RequestSerializationFailed
+    case Unknown
+}
+
+/// A custom serializer to handle XMLRPC responses
+final class WordPressOrgXMLRPCResponseSerializer: AFHTTPResponseSerializer
+{
+    override init() {
+        super.init()
+        self.acceptableContentTypes = ["application/xml", "text/xml"]
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    override func responseObjectForResponse(response: NSURLResponse?, data: NSData?, error: NSErrorPointer) -> AnyObject? {
+        let responseObject = super.responseObjectForResponse(response, data: data, error: error)
+        guard error.memory == nil,
+            let unwrappedData = data else {
+            return responseObject
+        }
+        let decoder = WPXMLRPCDecoder(data:unwrappedData)
+
+        guard !decoder.isFault(),
+            let responseXML = decoder.object() else {
+            let decoderError = decoder.error()
+            error.memory = decoderError
+            return responseObject
+        }
+
+        return responseXML
+    }
+}
+
+/// A custom serializer to handle XMLRPC requests
+final class WordPressOrgXMLRPCRequestSerializer: AFHTTPRequestSerializer
```

Comment:
In this case, the bulk of the request "serializing" is already being done by `WPXMLRPCEncoder` and `WordPressOrgXMLRPCRequestSerializer` feels like a lot of boilerplate to fit AFNetworking's model.

If you look at the documentation:
> The AFURLRequestSerialization protocol is adopted by an object that encodes parameters for a specified HTTP requests. Request serializers may encode parameters as query strings, HTTP bodies, setting the appropriate HTTP header fields as necessary.

But in the case of XML-RPC, we're not just encoding some arbitrary HTTP query parameters, but encoding a RPC method with its arguments. I think the request serializers were designed for RESTful APIs where the "method" is given by the request URL, and the parameters can be encoded in different ways. In this case, where both the method and parameters are part of the request body, we have to resort to hacks and boilerplate to make it work.

I agree with encapsulating the request creation inside a method (or a class), but wouldn't try to shoehorn it into `AFURLRequestSerialization`.